### ISSUE TITLE
[Question] Recording OpenGL minecraft window using MultiMC

### ISSUE THREAD

**Issue by Brottweiler** [MaartenBaert/ssr#289]
When I am recording regular Minecraft, I have minecraft autolaunched and that works fine. But when I use MultiMC, it doesn't work. I run autolaunch MultiMC, and when my game has started OpenGL.

This is the error:

```
[PageRecord::StartOutput] Error: Could not get the size of the OpenGL application. Either the application wasn't started correctly, or the application hasn't created an OpenGL window yet. If you want to start recording before starting the application, you have to enable scaling and enter the video size manually.
[PageRecord::StartOutput] Error: Something went wrong during initialization.
```

How can I select the proper window after the game has launched? I'd want to record using OpenGL injection and not a part of my screen. Thanks.

**Comment by StripedMonkey** [comment]
Sorry if this is a stupid question but what is multiMC? my guess from what
you posted MultiMC isn't a openGL program.

**Comment by Brottweiler** [comment]
@StripedMonkey It's a minecraft launcher. I wouldn't be using it unless I had to in this perticular case (playing and old version of the game because of an outdated modpack).

Minecraft is an OpenGL game.

**Comment by MaartenBaert** [comment]
OpenGL injection relies on a few environment variables (particularly LD_PRELOAD). The launcher may be removing these environment variables which will disable OpenGL injection. Does the launcher provide some way to change the command that will be executed, or modify environment variables?

How exactly are you starting MultiMC? From within SSR, or with the 'ssr-glinject' script?

**Comment by Brottweiler** [comment]
I am starting it from within SSR. It's a script, that looks like this: https://gist.github.com/470bd6eb9543c03b8f0b

I have these setting commands I can change:
![](http://i.imgur.com/Bp3CRvt.png)

**Comment by ubuntuaddicted** [comment]
this is very interesting to say the least. :)  i'll be watching this.  what distro or desktop are you using? what compositor? all this may or may not impact the injection (i think)

**Comment by Brottweiler** [comment]
Archlinux, Fluxbox, compton. I don't actually think this has anything to do with it. Regular minecraft works fine, it's just when I use MultiMC it doesn't "find" or "see" the OpenGL window.

**Comment by MaartenBaert** [comment]
MultiMC must be removing the environment variables then. Maybe it tries to start Minecraft from a clean environment.

Those 'pre-launch' and 'post-exit' commands won't do anything useful, you need to find a way to modify the actual command that is being executed (or the environment in which that command is executed). If the launcher does not provide a mechanism to do this, you may have to contact the developers.

One thing you could try is to create your own wrapper around the java program, something like this:
```
#!/bin/sh
ssr-glinject java "$@"
```
You could save this as 'java-glinject' and make it executable (```chmod +x java-glinject```), and then tell MultiMC to use this as a replacement for the real java executable.

**Comment by ubuntuaddicted** [comment]
this should be achievable pretty easily if i'm not mistaken. i currently launch my client using the following command/script.
[code]/home/ubu/My Games/minecraft/Minecraft.jar[/code]  that starts the launcher with my profile already loaded etc etc. so maybe you just need to ensure you're injecting at the correct level. within my ~/ folder, there's a .minecraft folder which has a bunch of scripts, one being launcher.jar.

does this info help you at all?

**Comment by MaartenBaert** [comment]
@ubuntuaddicted Brottweiler already said that normal Minecraft is working fine, he just needs a solution for MultiMC. There doesn't seem to be a way to change the actual command executed by MultiMC, however there is a way to change the path of the Java executable, so that's why I suggested the script.

**Comment by Brottweiler** [comment]
@MaartenBaert It worked! Thank you.

**Comment by ubuntuaddicted** [comment]
@Brottweiler what worked? the info that I provided? I'd like to know because I also want to use MultiMC and record locally with SSR. Glad you got it working.

**Comment by Brottweiler** [comment]
@ubuntuaddicted no, you weren't helpful at all! :wink: 

What worked was what @MaartenBaert suggested. I made a new file called `java-glinject` with this in it

```bash
#!/bin/sh
ssr-glinject java "$@"
```

Then in MultiMC I did this: http://i.imgur.com/Z1iqP3g.png

**Comment by ViktorMYT** [comment]
can u send me the file what is the file type i'm confused...

**Comment by Brottweiler** [comment]
It's a bash file. You literally just make a file called `java-glinject` or `java-glinject.sh` then put this in it

```bash
#!/bin/sh
ssr-glinject java "$@"
```

**Comment by ViktorMYT** [comment]
Thank's it works!
i made that file you told me without extension and i set it up as java
prefix in mc launcher and ssr does detect it and recording works aswell.
😀

пет, 20. мар 2020. у 22:11 Christoffer Tibell <notifications@github.com> је
написао/ла:

**Comment by ViktorMYT** [comment]
Hey,

I just signed the petition "Continue The Original Code Lyoko Series" and
wanted to see if you could help by adding your name.

Our goal is to reach 5,000 signatures and we need more support. You can
read more and sign the petition here:

http://chng.it/smyJrBZVSL

Thanks!
Viktor
